src/wasm/kotlin/src/arcs/Entity.kt

```diff
@@ -96,16 +96,49 @@ class StringDecoder(private var str: String) {
 class StringEncoder(private val sb: StringBuilder = StringBuilder()) {
 
     companion object {
-      fun encodeDictionary(dict: Map<String, String>): String {
-        val sb = StringBuilder()
-        sb.append(dict.size).append(":")
+        fun encodeDictionary(dict: Map<String, Any>): String {
+            val sb = StringBuilder()
+            sb.append(dict.size).append(":")
+
+            for((key, value) in dict) {
+                sb.append(key.length).append(":").append(key)
+                sb.append(encodeValue(value))
+            }
+            return sb.toString()
+        }
+
+        fun encodeList(list: List<Any>): String {
+          val sb = StringBuilder()
+          sb.append(list.size).append(":")
+          
+          for(value in list) {
+            sb.append(encodeValue(value))
+          }
+          return sb.toString()
+        }
 
-        for((key, value) in dict) {
-          sb.append(key.length).append(":").append(key)
-          sb.append(value.length).append(":").append(value)
+        fun encodeValue(value: Any): String {
```

Comment:
Since there aren't any loops within this method, consider using string templates instead of string builder, for readability. (they use StringBuilder under the covers for you)

e.g.

```kotlin
sb.append("T").append(value.length).append(":").append(value)
```
becomes
```kotlin
"T${value.length}:$value"
```